File: 2696-the-number-of-beautiful-subsets/the-number-of-beautiful-subsets.cpp
class Solution {
    int solve(vector<int>& nums, int k, int i, vector<int>& num) {
        if (i >= nums.size())
            return 0;
        num.push_back(nums[i]);
        int x = nums[i] + k;
        int a = 0;
        if (find(num.begin(), num.end(), nums[i] + k) == num.end() &&
            find(num.begin(), num.end(), nums[i] - k) == num.end()) {
                
            a = 1+solve(nums, k, i + 1, num);
        }
        num.pop_back();
        int b = solve(nums, k, i + 1, num);
        return a + b;
    }

public:
    int beautifulSubsets(vector<int>& nums, int k) {
        vector<int> num;
        return solve(nums, k, 0, num);
    }
};

File: 2696-the-number-of-beautiful-subsets/the-number-of-beautiful-subsets.java
class Solution {
    private int solve(int[] nums,int k,ArrayList<Integer> list,int i){
        if(i==nums.length) return 0;
        int p = nums[i];
        int a = 0;
        list.add(p);
        if(!list.contains(p+k) && !list.contains(p-k)){
            a = 1+solve(nums,k,list,i+1);
        }
        list.remove(list.size()-1);
        int b = solve(nums,k,list,i+1);
        return a+b;
    }
   
    public int beautifulSubsets(int[] nums, int k) {
        ArrayList<Integer> list = new ArrayList<>();
        return solve(nums,k,list,0);
    }
}